types/urijs/index.d.ts

```diff
@@ -239,8 +239,30 @@ declare namespace uri {
     type URITemplateCallback = (keyName: string) => URITemplateValue;
     type URITemplateInput = { [key: string]: URITemplateValue | URITemplateCallback } | URITemplateCallback;
 
+    type URITemplateLiteral = string;
+    interface URITemplateVariable {
+      name: string;
+      explode: boolean;
+      maxLength?: number;
+    }
+
+    interface URITemplateExpression {
+      expression: string;
+      operator: string;
+      variables: ReadonlyArray<URITemplateVariable>;
+    }
+
+    type URITemplatePart = URITemplateLiteral | URITemplateExpression;
+
     interface URITemplate {
       expand(data: URITemplateInput, opts?: Object) : URI;
+      parse(): this;
```

Comment:
Is it better to use `this` or `URITemplate`? I've always been explicit with the type so I'm not sure which is preferred here.

Reply:
Relatively new to TS here, so I can't comment authoritatively. I chose `this` here mainly because it feels like it implies a tad more about the *implementation* of the function, not only does it return an `URITemplate` type, it returns the same one you called `parse` on. Typing this as `parse(): URITemplate` is the same thing from a type declaration standpoint, but doesn't hint at the fact that it's specifically this one we're returning, not some other arbitrary URITemplate object.

That being said... Happy to change this if that's preferred style for this codebase. Empirically, I did find this:

```
[peter@respect:~/git/DefinitelyTyped] λ grep ": this;" types/**/*.d.ts | wc -l
5833
```

Reply:
That makes sense. I'll deffer to one of the maintainers on if it should be changed or not.